cmd/api/user_test.go

```diff
@@ -0,0 +1,927 @@
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusUnauthorized, w.Code, "HTTP Status Code")
+		var e APIEvent
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&e), "Decode JSON Event") {
+			expect.Equal("ERROR", e.LogLevel, "Event LogLevel")
+			expect.Equal(http.StatusUnauthorized, e.Code, "Event Code")
+			expect.Contains(e.Message, "unauthenticated", "Event Message")
+		}
+	}
+	// Get roles: unauthorized token (not an admin)
+	w = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/v1/user_roles", nil)
+	req.Header.Set("Authorization", "Bearer "+regularToken)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusForbidden, w.Code, "HTTP Status Code")
+		var e APIEvent
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&e), "Decode JSON Event") {
+			expect.Equal("ERROR", e.LogLevel, "Event LogLevel")
+			expect.Equal(http.StatusForbidden, e.Code, "Event Code")
+			expect.Contains(e.Message, "unauthorized", "Event Message")
+		}
+	}
+	// Get roles: happy path
+	w = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/v1/user_roles", nil)
+	req.Header.Set("Authorization", "Bearer "+adminToken)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusOK, w.Code, "HTTP Status Code")
+		var roles []string
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&roles), "Decode JSON Users") {
+			expect.Equal(1, len(roles))
+		}
+	}
+}
+
+func TestReadUserStatuses(t *testing.T) {
+	expect := assert.New(t)
+	// Get statuses: missing authentication token
+	w := httptest.NewRecorder()
+	req, err := http.NewRequest("GET", "/v1/user_statuses", nil)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusUnauthorized, w.Code, "HTTP Status Code")
+		var e APIEvent
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&e), "Decode JSON Event") {
+			expect.Equal("ERROR", e.LogLevel, "Event LogLevel")
+			expect.Equal(http.StatusUnauthorized, e.Code, "Event Code")
+			expect.Contains(e.Message, "unauthenticated", "Event Message")
+		}
+	}
+	// Get statuses: unauthorized token (not an admin)
+	w = httptest.NewRecorder()
+	req, err = http.NewRequest("GET", "/v1/user_statuses", nil)
+	req.Header.Set("Authorization", "Bearer "+regularToken)
+	req.Header.Set("Accept", "application/json;charset=UTF-8")
+	if expect.NoError(err) {
+		r.ServeHTTP(w, req)
+		expect.Equal(http.StatusForbidden, w.Code, "HTTP Status Code")
+		var e APIEvent
+		if expect.NoError(json.NewDecoder(w.Body).Decode(&e), "Decode JSON Event") {
+			expect.Equal("ERROR", e.LogLevel, "Event LogLevel")
+			expect.Equal(http.StatusForbidden, e.Code, "Event Code")
+			expect.Contains(e.Message, "unauthorized", "Event Message")
+		}
+	}
+	expect = assert.New(t)
```

Comment:
We could delete this redefinition of `expect`. The one above should be sufficient.